book/jvm-rebuilds-service.md

```diff
@@ -0,0 +1,167 @@
+# Java Virtual Machine Build Service
+
+## Overview
+
+The Java Virtual Machine Build Service (JBS, or JVM Build Service) is a controller that will rebuild Java and other JVM Language based dependencies from source.
+
+Java uses a binary distribution model, so the only way to ensure an application is completely built from source is to rebuild all its component libraries from source themselves. Unfortunately due to the nature of the Java ecosystem this is not an easy process and would normally require a large amount of manual toil.
+
+Although the Java binary distribution model is very convenient, it does mean that you are using untrusted community dependencies of unknown provenance. In general, you don't know who has uploaded these artifacts, or the environment that was used to build them. Building from source in a secure environment means that you can be be completly sure as to where the code you are running has come from.
+
+JBS automates this process as much as possible, making it much less time-consuming to create fully supply chain controlled Java builds.
+
+### Dependencies
+
+JBS depends on Tekton, but is otherwise largely independent of the rest of AppStudio. Given the right configuration it can be deployed outside of AppStudio, which is mainly used for testing and development.
+
+## Architecture
+
+### Flow Overview
+
+The JVM Build service can be enabled on a specific namespace by creating a `JBSConfig` object, with `enableRebuilds: true` specified. This will trigger the controller to create a deployment of the local Cache in the namespace. This cache has a number of purposes, but the primary one is to cache artifacts from Maven central to speed up Java builds.
+
+Once it is enabled and the cache has started the JVM Build Service will watch for `PipelineRun` objects with the `JAVA_COMMUNITY_DEPENDENCIES` results. This will be present on end user builds that have dependencies from untrusted upstream sources.
+
+When one of these `PipelineRun` objects is observed JBS will extract the dependency list, and attempt to rebuild all the listed artifacts from source.
```

Comment:
All listed artifacts or all listed JVM artifacts? Do we need to be more precise with the `artifact` term?